snowplow-tracker/src/main/java/com/snowplowanalytics/core/tracker/CrossDeviceParameter.kt

```diff
@@ -0,0 +1,29 @@
+package com.snowplowanalytics.core.tracker
+
+import com.snowplowanalytics.snowplow.controller.SessionController
+import com.snowplowanalytics.snowplow.controller.TrackerController
+
+/**
+ * The optional parameters to include in the query string added by [TrackerController.decorateLink]
+ */
+enum class CrossDeviceParameter {
+    /**
+     * Value of [SessionController.sessionId]
+     */
+    SESSION_ID,
+
+    /**
+     * Value of [Tracker.appId]
+     */
+    SOURCE_ID,
+
+    /**
+     * Value of [Tracker.platform]
+     */
+    SOURCE_PLATFORM,
+
+    /**
+     * Value of [Subject.userId]
+     */
+    USER_ID,
```

Comment:
I think we are missing the `reason` parameter here, which is the link text. That is a bit awkward one because we can't automatically populate based on the link as in the JS tracker, but will probably need to accept another optional string argument in the `decorateLink` call.

Reply:
Added a `reason` arg to `decorateLink`